src/components/container/ExternalAccountsLoadingIndicator/ExternalAccountsLoadingIndicator.tsx

```diff
@@ -0,0 +1,91 @@
+import { parseISO } from "date-fns/esm";
+import { useEffect, useState } from "react";
+import { useInterval } from "../../../hooks";
+import { LoadingIndicator } from "../../presentational";
+import "./ExternalAccountsLoadingIndicator.css"
+import language from "../../../helpers/language";
+
+export interface ExternalAccountsLoadingIndicatorProps {
+    previewState?: "externalAccountsFetchingData" | "externalAccountsLoaded"
+    externalAccountCategories?: string[];
+}
+
+let previewStateAccounts: ExternalAccount[] = [{
+    id: 1,
+    lastRefreshDate: add(new Date(), { days: -1 }).toISOString(),
+    status: "fetchingData",
+    provider:
+    {
+        "id": 37,
+        "name": "Cedars-Sinai Health System",
+        "category": "Provider",
+        "logoUrl": "https://mdhorg.ce.dev/api/v1/delegated/externalaccountproviders/37/logo"
+    }
+}];
+
+export default function (props: ExternalAccountsLoadingIndicatorProps) {
+    let [isWeb, setIsWeb] = useState<boolean>(false);
+    let [externalAccounts, setExternalAccounts] = useState<ExternalAccount[]>([]);
+
+    function refresh() {
+        if (props.previewState == "externalAccountsFetchingData") {
+            previewStateAccounts[0].status = "fetchingData";
+            setExternalAccounts(previewStateAccounts);
+            return;
+        }
+        if (props.previewState == "externalAccountsLoaded") {
+            previewStateAccounts[0].status = "fetchComplete";
+            setExternalAccounts(previewStateAccounts);
+            return;
+        }
+
+        MyDataHelps.getExternalAccounts().then(function (accounts) {
+            if (!!props.externalAccountCategories) {
+                accounts = accounts.filter(account => props.externalAccountCategories!.includes(account.provider.category));
+            }
+            accounts.forEach(account => {
+                if (account.status == 'fetchComplete' && isAfter(new Date(), add(parseISO(account.lastRefreshDate), { hours: 3 }))) {
+                    MyDataHelps.refreshExternalAccount(account.id);
+                    account.status = 'fetchingData';
+                }
+            });
+            setExternalAccounts(accounts);
+        });
+    }
+
+    function initialize() {
+        MyDataHelps.getDeviceInfo().then(function (deviceInfo) {
+            setIsWeb(deviceInfo.platform == "Web");
+        }).catch(function () {
+            setIsWeb(true);
+        }).finally(function () {
+            refresh();
+        });
+    }
+
+    useEffect(function () {
+        initialize();
+
+        MyDataHelps.on("externalAccountSyncComplete", refresh);
+        MyDataHelps.on("applicationDidBecomeVisible", refresh);
+        return () => {
+            MyDataHelps.off("externalAccountSyncComplete", refresh);
+            MyDataHelps.off("applicationDidBecomeVisible", refresh);
+        }
+    }, []);
+
+    let anyRefreshingAccounts = !!externalAccounts.find(e => e.status == "fetchingData");
+    useInterval(() => {
```

Comment:
No, hooks are never supposed to be used inside loops / conditions / nested functions: https://medium.com/@ryardley/react-hooks-not-magic-just-arrays-cd4f1857236e

Reply:
That makes sense in the `useState(...)` example they provided, because state is part of the re-rendering cycle.  I'm not sure I see how it would be a problem for `useInterval(...)` though, but I'm still pretty new to react 😃.

Reply:
I will be honest that I don't full understand it either, but I think that this is proper / conventional use of hooks at least?  I need to wrap my brain around it at a deeper level